pandas/tests/reshape/merge/test_merge.py

```diff
@@ -2167,7 +2169,7 @@ def test_merge_series(on, left_on, right_on, left_index, right_index, nm):
         name=nm,
     )
     expected = DataFrame(
-        {"A": [2, 4], "B": [1, 3]},
+        {"A": [2, 4], nm: [1, 3]},
```

Comment:
Could you please add a dedicated test for this?

Reply:
I'll add one